packages/shared/src/components/filters/FeedFilters.tsx

```diff
@@ -12,46 +20,63 @@ interface FeedFiltersProps {
   onBack?: () => void;
 }
 
-let timeout: NodeJS.Timeout;
-
 export default function FeedFilters({
   isOpen,
   onBack,
 }: FeedFiltersProps): ReactElement {
+  const { alerts, updateAlerts } = useContext(AlertContext);
+  const { feedSettings } = useFeedSettings();
   const [hidden, setHidden] = useState(true);
+  const timeoutRef = useRef<number>();
 
   useEffect(() => {
     if (isOpen) {
-      if (timeout) clearTimeout(timeout);
+      if (
+        alerts?.filter &&
+        (feedSettings?.includeTags?.length ||
+          feedSettings?.blockedTags?.length ||
+          feedSettings?.excludeSources?.length ||
+          feedSettings?.advancedSettings?.length)
+      ) {
+        updateAlerts({ filter: false });
+      }
+
+      if (!hidden) {
+        return;
+      }
+
+      if (timeoutRef.current) clearTimeout(timeoutRef.current);
       setHidden(false);
       return;
     }
 
-    timeout = setTimeout(() => setHidden(true), 300);
-  }, [isOpen]);
+    if (hidden) {
+      return;
+    }
+
+    timeoutRef.current = window.setTimeout(() => setHidden(true), 300);
```

Comment:
We have to clear timeout when done
```suggestion
    timeoutRef.current = window.setTimeout(() => setHidden(true), 300);
    return () => timeoutRef.current && clearTimeout(timeoutRef.current);
```